tests/apache/livy/triggers/test_livy.py

```diff
@@ -0,0 +1,175 @@
+
+    task = [i async for i in trigger.run()]
+    assert len(task) == 1
+    assert (
+        TriggerEvent(
+            {
+                "status": "error",
+                "batch_id": 1,
+                "response": "Batch 1 did not succeed with Cannot connect to host livy:8998 ssl:default "
+                "[Name or service not known]",
+                "log_lines": None,
+            }
+        )
+        in task
+    )
+
+
+@pytest.mark.asyncio
+async def test_livy_trigger_poll_for_termination_with_client_error():
+    """
+    Test if the poll_for_termination polls for batch with client error
+    :return:
+    """
+    trigger = LivyTrigger(batch_id=1, spark_params={}, livy_conn_id="livy_default", polling_interval=30)
+
+    with pytest.raises(ClientConnectionError):
+        await trigger.poll_for_termination(1)
+
+
+@pytest.mark.asyncio
+@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.get_batch_state")
+@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.dump_batch_logs")
+async def test_livy_trigger_poll_for_termination_success(mock_dump_batch_logs, mock_get_batch_state):
+    """
+    Test if the poll_for_termination polls for batch to terminate for success
+    :return:
+    """
+    mock_get_batch_state.return_value = {"batch_state": BatchState.SUCCESS}
+    mock_dump_batch_logs.return_value = ["mock_log"]
+    trigger = LivyTrigger(batch_id=1, spark_params={}, livy_conn_id="livy_default", polling_interval=30)
+
+    task = await trigger.poll_for_termination(1)
+
+    assert task == {
+        "status": "success",
+        "batch_id": 1,
+        "response": "Batch 1 succeeded",
+        "log_lines": ["mock_log"],
+    }
+
+
+@pytest.mark.asyncio
+@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.get_batch_state")
+@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.dump_batch_logs")
+async def test_livy_trigger_poll_for_termination_error(mock_dump_batch_logs, mock_get_batch_state):
+    """
+    Test if the poll_for_termination polls for batch to terminate with error.
+    :return:
+    """
+    mock_get_batch_state.return_value = {"batch_state": BatchState.ERROR}
+    mock_dump_batch_logs.return_value = ["mock_log"]
+    trigger = LivyTrigger(batch_id=1, spark_params={}, livy_conn_id="livy_default", polling_interval=30)
+
+    task = await trigger.poll_for_termination(1)
+
+    assert task == {
+        "status": "error",
+        "batch_id": 1,
+        "response": "Batch 1 did not succeed",
+        "log_lines": ["mock_log"],
+    }
+
+
+@pytest.mark.asyncio
+@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.get_batch_state")
+@mock.patch("astronomer.providers.apache.livy.hooks.livy.LivyHookAsync.dump_batch_logs")
+async def test_livy_trigger_poll_for_termination_state(mock_dump_batch_logs, mock_get_batch_state):
+    """
+    Test if the poll_for_termination polls for batch to terminate with error.
+    :return:
```

Comment:
?

Reply:
`Test if the poll_for_termination() in the triggerer returned error response when get_batch_state() failed.`

Changing it in the docstring.

Reply:
[Done](https://github.com/astronomer/astronomer-providers/pull/178/commits/132fa197b35937edf9d07a1eac1d94a0350248fd)